tools/clang/tools/dxil2spv/lib/dxil2spv.cpp

```diff
@@ -713,13 +714,32 @@ const spirv::SpirvType *Translator::toSpirvType(llvm::StructType *structType) {
 
 spirv::SpirvInstruction *
 Translator::getSpirvInstruction(llvm::Value *instruction) {
-  spirv::SpirvInstruction *spirvInstruction = instructionMap[instruction];
-  if (!spirvInstruction) {
-    emitError("Expected SPIR-V instruction not found for DXIL instruction: %0",
-              *instruction);
+  if (!instruction) {
+    emitError("Unexpected null DXIL instruction");
     return nullptr;
   }
-  return spirvInstruction;
+
+  if (spirv::SpirvInstruction *spirvInstruction = instructionMap[instruction])
+    return spirvInstruction;
+
+  if (llvm::Constant *constant = llvm::dyn_cast<llvm::Constant>(instruction)) {
```

Comment:
nit: the type on the LHS is obvious, so we could do `if (auto *constant = ) { `

Reply:
we could also drop braces

Reply:
Done